### PR TITLE
Reporting features

### LINKED ISSUE #196: Reporting Features
@dting 

Posting this so we can keep track of our progress. Let's keep this issue open until all reporting features are complete:

- [x] Home Visits Outstanding
- [x] Disaggregate children who are in Child First Authority afteschool program and their trends
- [x] List of interventions by student, selected students, or cumulative
- [x] List of students who are chronically absent at any time of the year (missed more than 20 days)
- [x] Phone calls outstanding
- [x] Letters outstanding
- [x] SST referrals outstanding
- [x] Court Referrals outstanding

### PR DESCRIPTION
**Pull request by pdotsani**
@dting 

resolves #196 

basically a repeat of dashboard page, with some small tweaks

updated outreach model, switch data type to enum
created school-reports route
created current and at risk absence report queries

### CODE REVIEW

**Comment by pdotsani** [comment]
@dting  

Is there a place in our app to customize triggers for notifications? I'm thinking we could add some functionality in the admin section that allows teachers to customize when they get notified for SST, phone calls, etc. since each school will have different rules as to how these events are triggered. This seems like more of a necessity after taking a closer look at the reporting functionality.

This could also resolve the concern about inserting a specific formula into the query to get the ARCA in  the `absencerecord.controller`, they could customize this as well.

**Comment by dting** [comment]
Ya we need to add something to allow teachers to customize the triggers for their school. I think that might be the next thing i tackle. Should be pretty straight forward

**Comment by pdotsani** [comment]
Sounds good. I'll create an issue and you can take it whenever you're free.

**Comment by dting** [comment]
Hmm. I don't think this should have been merged. You removed the enum from the wrong file, interventions rather than outreaches. I'll make a change list that reverts that change.

File: client/app/main/school-reports/school-reports.controller.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+var app = angular.module('app');
+
+function SchoolReports($scope, Auth, uiGridGroupingConstants,
+	Student, AbsenceRecord) {
```
Comment (dting) [inline client/app/main/school-reports/school-reports.controller.js]: Formating error not fixed.

File: server/api/absence-record/absence-record.controller.js [hunk 2]
```diff
@@ -61,6 +61,56 @@ exports.current = function (req, res) {
...
+    if (err) return handleError(res, err);
+    AbsenceRecord.populate(results, 'school entries.student',
+      function (err, entries) {
+        if (err) return handleError(res, err);
+        entries = _.filter(entries, 
+          function(entry){
+            var present = entry.entries.present;
+            var enrolled = entry.entries.enrolled;
+            var tot = ( present / enrolled ).toFixed(2);
+            return tot === 0.9;
```
Comment (dting) [inline server/api/absence-record/absence-record.controller.js]: This should be a less than comparison if anything. For enrollment to be exactly 90 percent is a fairly strange criteria.

**Pull request merged**